### PR TITLE
Make eq_translations a python package

### PR DESCRIPTION
**Pull request by bitdivision**
This PR handles converting the eq-translations project into a python package. This will allow us to use the translations functionality from within a docker container without running docker in docker or something similar.

Currently, I have not dealt with uploading to pypi. For the moment, I think it seems reasonable to create releases (tags) on this repo which can be referenced from pip when it is installed from github. If people feel that this should go on pypi then I can add a script to upload it, though my opinion is that it shouldn't.

I've added some linting to help keep the code consistent.

The directory structure has been changed a little to support building a python package, and I've refactored the CLI scripts to call python functions which can also be used elsewhere. These functions just handle creating class instances etc (currently in `entrypoints.py`).

The CLI scripts are now entrypoints, so once the package is installed, they can be used like: `translate_survey ...` or `compare_schemas ...`.

You can test the package by installing it with `pip install .` from the root directory (setup a virtualenv and run `pipenv shell` first). 

Should be reviewed in tandem with: https://github.com/ONSdigital/eq-survey-runner/pull/2240

### CODE REVIEW

**Comment by bitdivision** [comment]
Will work on updating the README now.

File: README.md [hunk 1]
```diff
@@ -12,31 +13,50 @@ pip install --upgrade pip setuptools pipenv
...
 ```
+This will output the translatable text to an POT file.
 
-To translate all surveys in a directory run:
+
+After the text has been translated, create a new translated survey with:
 
 ```
-pipenv run python -m cli.translate_all_surveys <top_level_schema_directory>
+pipenv run python -m eq_translations.cli.translate_survey <schema_file> <translation_path> <output_directory>
```
Comment (JamesGardiner) [inline README.md]: Is the `python -m eq_translations.cli.` part not redundant now that these are registered inside console scripts?

Comment (bitdivision) [reply README.md]: They're registered as entry points if you install the package, but won't be available if you don't.

I can remove if we feel this pollutes the readme, but I figured it doesn't do any harm at the moment.

Comment (JamesGardiner) [reply README.md]: No that's a fair point. Maybe make it explicit that these could be in two states (installed vs not installed), and that when installed they will be shimmed in the PATH

Comment (bitdivision) [reply README.md]: Done

File: eq_translations/entrypoints.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+    translation = SchemaTranslation()
+    translation.load(translation_path)
+
+    translated_schema = survey_schema.translate(translation)
+    translated_schema.save(os.path.join(output_directory, schema_name))
+
+    compare_schemas(survey_schema.schema, translated_schema.schema)
+
+def handle_compare_schemas(source_schema, target_schema):
```
Comment (ajmaddaford) [inline eq_translations/entrypoints.py]: I think the entrypoint names should be consistent, either all prefixed with `handle_` or none.

File: eq_translations/cli/template_extractor.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        help='The destination directory for the translation template'
+    )
+
+    args = parser.parse_args()
+
+    if not os.path.isdir(args.OUTPUT_DIRECTORY):
+        print('Output directory does not exist')
+        exit(2)
+
+    extract_template(args.SCHEMA_PATH, args.OUTPUT_DIRECTORY)
```
Comment (ajmaddaford) [inline eq_translations/cli/template_extractor.py]: I think we should be more explicit about the potential exit codes for all of the script actions.

Comment (bitdivision) [reply eq_translations/cli/template_extractor.py]: Looking at this, python will exit with code `1` if an unhandled exception is raised.

Possibly valuable would be to add specific handling for bad file paths etc. But given that there aren't defined exit codes for those issues, I think I will leave this as is for the moment. The traceback and exception will give a fair amount of information to resolve it.

**Pull request merged**